### PR TITLE
[Android] Use scanCode to detect 102nd key

### LINKED ISSUE #1179: [Android] 102nd (oE2) Key not responding properly in KMEA
K_oE2 on the French keyboard contains '<' and '>' and my keyboard uses it. I'm noticing that even though KMWeb correctly handles the K_oE2 key (and it exists in the .js file), Android hardware treats it as a duplicate of the K_BKSLASH key and outputs the result of the K_BKSLASH Key instead. 

From the resources at:
https://source.android.com/devices/input/keyboard-devices
it looks like Android reports several "international" keys as KEYCODE_BACKSLASH, so you may not be getting enough information to solve this bug. Maybe you can ask Android to split them out in later versions? 

Secondarily, is there a plan to make all Keyman Keycodes in caps, like oE2 becoming OE2?

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #1179 

Reference: https://source.android.com/devices/input/keyboard-devices

Since Android overloads the 102nd key with KEYCODE_BACKSLASH, use the scan code to detect the 102nd key. Pass the code `226` for KMW to process.

### CODE REVIEW

**Comment by darcywong00** [comment]
### Testing configuration
Android 8.1.0 device with physical AZERTY keyboard
* Settings - set the AZERTY physical keyboard to "English (US)"

Keyman app
* Install French-> French Keyman keyboard.

In the Keyman app,
- Type `*` (normally `\`) and verify generates `*`
- Type 102nd key and verify generates `<` 
- Type shfit-102nd key and verify generate `>`

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java [hunk 1]
```diff
@@ -161,8 +167,15 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
       return false;
     }
 
+    int code;
+    if ((keyCode == KeyEvent.KEYCODE_BACKSLASH ) && (event.getScanCode() == SCAN_CODE_KEY_102ND) ) {
```
Comment (darcywong00) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java]: Note: If the user sets their physical AZERTY keyboard to French, 
the 102nd key generates `KeyEvent.KEYCODE_PLUS` (scan code also = `SCAN_CODE_KEY_102ND`)

**Comment by mcdurdin** [comment]
See also the following quote:

The Android API sometimes refers to the Linux key code associated with a key as its "scan code". This is technically incorrect in but it helps to distinguish Linux key codes from Android key codes in the API.

from https://source.android.com/devices/input#understanding-hid-usages-and-event-codes

**Comment by darcywong00** [comment]
Remapped to use Linux Key Codes (LKC / "scan code") to KMW codes.

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java [hunk 2]
```diff
@@ -101,6 +104,134 @@
...
+    '4',    //        public static final int KEY_4 = 0x05;
+    '5',    //        public static final int KEY_5 = 0x06;
+    '6',    //        public static final int KEY_6 = 0x07;
+    '7',    //        public static final int KEY_7 = 0x08;
+    '8',    //        public static final int KEY_8 = 0x09;
+    '9',    //        public static final int KEY_9 = 0x0A;
+    '0',    //        public static final int KEY_0 = 0x0B;
+    189,    //        public static final int KEY_MINUS = 0x0C;
+    187,    //        public static final int KEY_EQUALS = 0x0D;
+    0,      //        public static final int KEY_BACKSPACE = 0x0E;
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java]: For backspace, tab, and enter, I kept their mappings previously defined as:
```
0,      //        public static final int KEYCODE_TAB = 61;
13,     //        public static final int KEYCODE_ENTER = 66;
0,      //        public static final int KEYCODE_DEL = 67;
```

I can add KMW mappings for Backspace and Delete.

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java [hunk 3]
```diff
@@ -101,6 +104,134 @@
...
+    'R',    //        public static final int KEY_R = 0x13;
+    'T',    //        public static final int KEY_T = 0x14;
+    'Y',    //        public static final int KEY_Y = 0x15;
+    'U',    //        public static final int KEY_U = 0x16;
+    'I',    //        public static final int KEY_I = 0x17;
+    'O',    //        public static final int KEY_O = 0x18;
+    'P',    //        public static final int KEY_P = 0x19;
+    219,    //        public static final int KEY_LEFTBRACE = 0x1A;
+    221,    //        public static final int KEY_RIGHTBRACE = 0x1B;
+    13,     //        public static final int KEY_ENTER = 0x1C;
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java]: I've added mappings for Tab and Backspace, but still need to test them when I'm back in the office Monday.

[the remark this answers is not in the document]
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMHardwareKeyboardInterpreter.java]: I made a test keyboard and confirmed Tab, Backspace, Enter, and the 102nd key are mapped to KMW
```
+ [K_BKSP] > '[GO BACK]'
+ [K_ENTER] > '[YOU MAY ENTER]'
+ [K_TAB] > '[TAB]'
+ [K_OE2] > '[102nd key]'

```

**Comment by erros84** [comment]
Great, which version of KMEA will contain this fix?

**Comment by mcdurdin** [comment]
This will be in KMEA 11.0.

**Pull request merged**